src/allocation_engine/interval_tree.rs

```diff
@@ -344,7 +344,7 @@ impl InnerNode {
                     .ok_or(Error::Overflow)?;
                 // We should check that starting from the desired address the
                 // whole memory slot will fit in the selected node.
-                if end_address > node.key.end() {
+                if end_address.checked_sub(1).ok_or(Error::Underflow)? > node.key.end() {
```

Comment:
It looks like ranges in the crate are inconsistently defined:
- in this case, the range is non inclusive on the right end
- but in the `Range` definition, ranges are inclusive on both sides.

This is making the code hard to reason about, as you don't know out of the box with what kind of range you're dealing with.

Reply:
Chatted offline, resolving this. It looks like we're using Inclusive ranges everywhere. To set clear expectations about this, we're going to rename the structure from Range to RangeInclusive, thus also following the Rust std::ops::Range* defines.